src/reactviews/pages/ExecutionPlan/executionPlanPage.tsx

```diff
@@ -33,10 +34,17 @@ const useStyles = makeStyles({
     },
 });
 
-export const ExecutionPlanPage = () => {
+interface ExecutionPlanPageProps {
+    context?: QueryResultState;
+}
+
+export const ExecutionPlanPage: React.FC<ExecutionPlanPageProps> = ({
+    context,
+}) => {
     const classes = useStyles();
-    const provider = useContext(ExecutionPlanContext);
-    const loadState = provider?.state?.loadState ?? ApiStatus.Loading;
+    const provider = context ? context : useContext(ExecutionPlanContext);
```

Comment:
I thought the point of `useContext` was to remove the need to pass the context in as a property.  @aasimkhan30?

No need to change for this PR since this isn't just a nit, but something we should circle back to.